src/rss_etl.py

```diff
@@ -0,0 +1,175 @@
+from supabase import create_client, Client
+import feedparser
+import os
+import re
+from enum import Enum
+from dataclasses import dataclass
+
+"""
+File for extracting data from the RSS feeds of the PA Senate and House of Representatives
+and loading it into Supabase.
+
+The database tables updated by this script are:
+    Bills
+    Revisions
+"""
+
+# from dotenv import load_dotenv
+# load_dotenv()  # for local secret management with .env file
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+# sb_api_url = os.getenv("SUPABASE_API_URL")  # local secret management with .env file
+# sb_api_key = os.getenv("SUPABASE_API_KEY")  # local secret management with .env file
+sb_api_url = os.environ["SUPABASE_API_URL"]  # github actions secret management
+sb_api_key = os.environ["SUPABASE_API_KEY"]  # github actions secret management
+
+
+class LegislativeBody(Enum):
+    SENATE = "Senate"
+    HOUSE = "House"
+
+
+REVISIONS_OUTPUT_DICT = {
+    "bill_internal_id": "",
+    "printer_no": "",
+    "full_text_link": "",
+    "publication_date": "",
+    "revision_guid": "",
+    "description": "",
+}
+
+BILL_OUTPUT_DICT = {
+    "bill_number": "",
+    "legislative_session": "",
+    "session_type": "",
+    "chamber": "",
+    "legislative_id": "",
+}
+
+
+class BillIdentifier:
+    """
+    Extracts unique information from a bill's guid and embeds them in discrete properties
+    """
+
+    @staticmethod
+    def extract_bill_guid(revision_guid):
+        return re.search("BillGuid=(.*)", revision_guid).group(1)
+
+    @staticmethod
+    def determine_chamber(chamber_letter):
+        if chamber_letter == "H":
+            return "House"
+        elif chamber_letter == "S":
+            return "Senate"
+        else:
+            raise Exception("Invalid chamber letter in guid")
+
+    def __init__(self, guid):
+        self.guid = guid
+        self.bill_guid = re.search(r"(\d{4}\d[H|S]B\d+)P\d+", guid).group(1)
```

Comment:
For this regex; just add comments describing what it's doing and give an example